lib/script/script.js

```diff
@@ -713,14 +727,44 @@ Script.fromAddress = function(address) {
 
 /**
  * @param {Network=} network
- * @return {Address} the associated address for this script
+ * @return {Address|boolean} the associated address information object
+ * for this script if any, or false
+ */
+Script.prototype.getAddressInfo = function() {
+  var Address = require('../address');
+  var info = {};
+  if (this.isScriptHashOut()) {
+    info.hashBuffer = this.getData();
+    info.type = Address.PayToScriptHash;
+  } else if (this.isPublicKeyHashOut()) {
+    info.hashBuffer = this.getData();
+    info.type = Address.PayToPublicKeyHash;
+  } else if (this.isPublicKeyHashIn()) {
+    // hash the publickey found in the scriptSig
+    info.hashBuffer = Hash.sha256ripemd160(this.chunks[1].buf);
```

Comment:
Thought about it, but I think it would be weird to return the hash of the public key, given the input script in p2pkh case contains a public key AND a signature... I mean, why would getData only return one of those 2, and hashed?